operations.go

```diff
@@ -285,3 +285,36 @@ func (r *KillProcessOp) Operate(t testing.TB) {
 		t.Fatalf("Operate(t) on %s: %v", r, err)
 	}
 }
+
+
+// NewSwitchControlProcessor creates a new switch control processor operation.
+func (o *Operations) NewSwitchControlProcessor() *SwitchControlProcessorOp {
+	return &SwitchControlProcessorOp{
+		dev: o.dev,
+	}
+}
+
+// SwitchControlProcessorOp is an operation that switch from current route procesor
+// to provided route processor.
+type SwitchControlProcessorOp struct {
+	dev binding.Device
+	dest string
+}
+
+func (s *SwitchControlProcessorOp) String() string {
+	return fmt.Sprintf("SwitchControlProcessorOp%+v", *s)
+}
+
+// WithDestinationControlProcessor sets the destination route processor to switch to.
+func (s *SwitchControlProcessorOp) WithDestinationControlProcessor(dest string) *SwitchControlProcessorOp {
```

Comment:
naming this "WithDestination" is enough. The context is already the SwitchControlProcessor operation, so the added "ControlProcessor" here is a bit redundant.